samigo/samigo-qti/src/java/org/sakaiproject/tool/assessment/qti/helper/assessment/AssessmentHelperBase.java

```diff
@@ -456,17 +459,30 @@ public void updateIPAddressSet(Assessment assessmentXml,
   public void updateAttachmentSet(Assessment assessmentXml, Set attachmentSet)
   {
     Iterator iter = attachmentSet.iterator();
-    AttachmentData attachmentData = null;
     StringBuffer attachment = new StringBuffer();
     while (iter.hasNext())
     {
-    	attachmentData = (AttachmentData) iter.next();
-    	attachment.append(attachmentData.getResourceId().replaceAll(" ", ""));
-    	attachment.append("|");
-    	attachment.append(attachmentData.getFilename());
-    	attachment.append("|");
-    	attachment.append(attachmentData.getMimeType());
-    	attachment.append("\n");
+        Object elt = iter.next();
+
+        if (elt instanceof AttachmentData) {
+            AttachmentData attachmentData = (AttachmentData) elt;
+            attachment.append(attachmentData.getResourceId().replaceAll(" ", ""));
```

Comment:
Lines from 469 to 474 are the same 477 to 482
Better put them out of the if

Reply:
attachmentData is a different type in the two if cases.

Reply:
Seems like that would be more lines of code in total, so not sure it's worth it just to get rid of what looks likes fairly trivial duplication.